src/main/java/gregtech/api/items/toolitem/behavior/IToolBehavior.java

```diff
@@ -0,0 +1,143 @@
+package gregtech.api.items.toolitem.behavior;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.util.ITooltipFlag;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * Describes generic behaviour attachable to tools. Multiple behaviours can be attached to one tool.
+ */
+public interface IToolBehavior {
+
+    /**
+     * @param stack    The current ItemStack
+     * @param target   the entity being hit
+     * @param attacker the entity hitting the other
+     */
+    default void hitEntity(@Nonnull ItemStack stack, @Nonnull EntityLivingBase target, @Nonnull EntityLivingBase attacker) {
+
+    }
+
+    /**
+     * Called before a block is broken. Return true to prevent default block harvesting.
```

Comment:
This java doc needs updating. It states return true to prevent block harvesting, but the method is void.